exception_reports/logs.py

```diff
@@ -19,39 +18,21 @@ def async_exception_handler(loop, context):
 
 
 class AddExceptionReportFilter(logging.Filter):
-    def __init__(self, storage_backend=LocalErrorStorage(), output_html=True, output_json=False):
+    def __init__(self, storage_backend=LocalErrorStorage(), output_format='json'):
         super().__init__()
         self.storage_backend = storage_backend
-        self.output_html = output_html
-        self.output_json = output_json
-        self.enabled = output_json or output_html
+        self.output_format = output_format
 
     def filter(self, record):
-        if not self.enabled:
-            return True
-
         if record.levelno >= logging.ERROR:
-            if not getattr(record, '_exception_data', None):
-                record._exception_data = None
-            exc_info = record.exc_info or (None, record.getMessage(), get_logger_traceback())
-            try:
-                record._exception_data = ExceptionReporter(*exc_info).get_traceback_data()
-            except Exception as e:
-                logger.warning(f"Error getting traceback data {repr(e)}")
-
             if not getattr(record, 'data', None):
                 setattr(record, 'data', {})
+            exc_type, exc_value, tb = record.exc_info or (None, record.getMessage(), get_logger_traceback())
 
-            if record._exception_data:
-                if self.output_html:
-                    html = render_exception_report(record._exception_data)
-                    filename = gen_error_filename(extension='html')
-                    record.data['error_report'] = self.storage_backend.write(filename, html)
-
-                if self.output_json:
-                    json_str = render_exception_json(record._exception_data)
-                    filename = gen_error_filename(extension='json')
-                    record.data['error_report_json'] = self.storage_backend.write(filename, json_str)
+            try:
+                record.data['error_report'] = create_exception_report(exc_type, exc_value, tb, self.output_format, self.storage_backend)
+            except Exception as e:
+                logger.warning(f"Error generating exception report {repr(e)}")
```

Comment:
raise again so we don't quietly swallow an error in create_exception_report?

Reply:
This code deals with error level log records.  As long as we return `True`, the log will be written.  I think it's appropriate that this exception, thrown by exception report library, be swallowed so that the original log message can be written.  

Basically we can't have exception reports being created for problems in the exception reports library since that would permit infinite recursion.

Reply:
Yes. It would try to create an exception report, which could work, but could end up throwing the same exception.  Had this happen a few times before with datatypes that it didn't know how to render.

Its true that this doesn't surface problems with exception reports very well.   I think an error handling library is intended to make systems more stable and easier to debug.  It has to "do no harm".  The absolute worst case scenario is for this library to contribute to system instability.  

For that reason, I'm willing to make this tradeoff.  If there is a way to have everything, great! I don't see a simple path for that though.